GitUI/RepositoryCurrentBranchNameProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿using GitCommands;
+using GitCommands.Git;
+
+namespace GitUI
+{
+    public interface IRepositoryCurrentBranchNameProvider
+    {
+        string GetCurrentBranchName(string repositoryPath);
+    }
+
+    public sealed class RepositoryCurrentBranchNameProvider : IRepositoryCurrentBranchNameProvider
+    {
+        public string GetCurrentBranchName(string repositoryPath)
+        {
+            if (!AppSettings.ShowRepoCurrentBranch)
+            {
+                return string.Empty;
+            }
+
+            string branchName = GitModule.GetSelectedBranchFast(repositoryPath);
+            if (string.IsNullOrWhiteSpace(branchName) || branchName == DetachedHeadParser.DetachedBranch)
+            {
+                branchName = $"({Strings.NoBranch})";
```

Comment:
Either wrap all branches in brackets or none.
But we can emphasize that "no branch" is not a branch by using angle brackets.
```suggestion
                branchName = $"<{Strings.NoBranch}>";
```

Reply:
Everywhere else we have it as "(no branch)".

Reply:
> Everywhere else we have it as "(no branch)".

This is not the full truth. For the submodules in the ROT, all branches are appended in round brackets, i.e. there it is "no branch", else it would have been displayed as "((no branch))".

Reply:
The `Dashboard`, `FormCommit` or ROT do not currently use the same functionality, this is outside the scope of this PR, but it is certainly something to do.

Reply:
The easiest and correct solution is to change `Strings.NoBranch`.
This does not need to be done in this PR.

Reply:
This is a reason why I preferred to not do this change. It seems like "no branch" is the name of the branch. If you want to display the string (I would prefer to not see something that do not exist, second opinion), it must be more unique. Maybe [no branch]

Reply:
Once we have unified "no branch" handling it will be easy to change, and we can change it to whatever we agree.
Right now I want to have a consistency.